server/src/main/java/com/google/musicanalysis/site/YoutubeGenres.java

```diff
@@ -68,12 +74,17 @@ protected void calculateMusicCount(JsonArray videos) {
             }
         }
 
-        if (isMusic && totalSubgenres == 0) {
+        if (isMusic) {
+          // likedMusicHistory records video numbers that are music
+          likedMusicHistory.add(firstVideoCount + i);
+
+          if (totalSubgenres == 0) {
             // video only classified as Music so we update as "Other music"
-            this.updateGenre("Other music");
+            this.updateGenre("Other music");            
+          }
         }
     }
-    return;
+    return firstVideoCount + videos.size();
```

Comment:
nit: Rather than calling `videos.size()` twice, it might be more efficient to store that in a variable and use it throughout. I think technically this is okay because `.size()` is constant? (not sure) but if it takes O(n) and you're not modifying the array there's no need to call this method twice